samples/dnn/virtual_try_on.py

```diff
@@ -0,0 +1,435 @@
+
+    points_X_batch = np.expand_dims(points[:,:,:,0], axis=3)
+    points_Y_batch = np.expand_dims(points[:,:,:,1], axis=3)
+
+    if points_b == 1:
+        points_X_batch = expand_torch(points_X_batch, (batch_size, ) + points_X_batch.shape[1:])
+        points_Y_batch = expand_torch(points_Y_batch, (batch_size, ) + points_Y_batch.shape[1:])
+
+    points_X_prime = A_X[:,:,:,:,0]+ \
+                    np.multiply(A_X[:,:,:,:,1], points_X_batch) + \
+                    np.multiply(A_X[:,:,:,:,2], points_Y_batch) + \
+                    np.sum(np.multiply(W_X, expand_torch(U, W_X.shape)), 4)
+
+    points_Y_prime = A_Y[:,:,:,:,0]+ \
+                    np.multiply(A_Y[:,:,:,:,1], points_X_batch) + \
+                    np.multiply(A_Y[:,:,:,:,2], points_Y_batch) + \
+                    np.sum(np.multiply(W_Y, expand_torch(U, W_Y.shape)), 4)
+
+    return np.concatenate((points_X_prime, points_Y_prime), 3)
+
+def postprocess(theta):
+    grid_X, grid_Y, N, P_X, P_Y = prepare_to_transform()
+    warped_grid = apply_transformation(theta, np.concatenate((grid_X, grid_Y), axis=3), N, P_X, P_Y)
+    return warped_grid
+
+def bilinear_sampler(img, grid):
+    x, y = grid[:,:,:,0], grid[:,:,:,1]
+
+    H = img.shape[2]
+    W = img.shape[3]
+    max_y = H - 1
+    max_x = W - 1
+
+    # rescale x and y to [0, W-1/H-1]
+    x = 0.5 * (x + 1.0) * (max_x - 1)
+    y = 0.5 * (y + 1.0) * (max_y - 1)
+
+    # grab 4 nearest corner points for each (x_i, y_i)
+    x0 = np.floor(x).astype(int)
+    x1 = x0 + 1
+    y0 = np.floor(y).astype(int)
+    y1 = y0 + 1
+
+    # calculate deltas
+    wa = (x1 - x) * (y1 - y)
+    wb = (x1 - x) * (y  - y0)
+    wc = (x - x0) * (y1 - y)
+    wd = (x - x0) * (y  - y0)
+
+    # clip to range [0, H-1/W-1] to not violate img boundaries
+    x0 = np.clip(x0, 0, max_x)
+    x1 = np.clip(x1, 0, max_x)
+    y0 = np.clip(y0, 0, max_y)
+    y1 = np.clip(y1, 0, max_y)
+
+    # get pixel value at corner coords
+    img = img.reshape(-1, H, W)
+    Ia = img[:, y0, x0].swapaxes(0, 1)
+    Ib = img[:, y1, x0].swapaxes(0, 1)
+    Ic = img[:, y0, x1].swapaxes(0, 1)
+    Id = img[:, y1, x1].swapaxes(0, 1)
+
+    wa = np.expand_dims(wa, axis=0)
+    wb = np.expand_dims(wb, axis=0)
+    wc = np.expand_dims(wc, axis=0)
+    wd = np.expand_dims(wd, axis=0)
+
+    # compute output
+    out = wa*Ia + wb*Ib + wc*Ic + wd*Id
+    return out
+
+
+if __name__ == "__main__":
+    pose = get_pose_map(args.input_image, findFile(args.openpose_proto),
+                        findFile(args.openpose_model), args.backend, args.target)
+    segm_image = parse_human(args.input_image, args.segmentation_model)
+    segm_image = cv.resize(segm_image, (192, 256), cv.INTER_LINEAR)
+
+    agnostic = prepare_agnostic(segm_image, args.input_image, pose)
+    warped_cloth = get_warped_cloth(args.input_cloth, agnostic, args.gmm_model, args.backend, args.target)
+    output = get_tryon(agnostic, warped_cloth, args.tom_model, args.backend, args.target)
```

Comment:
Maybe better to replace functions to classes? Networks will be initialized in `__init__` and there would be processing methods such `prepare_agnostic`.